core/src/main/java/de/danoeh/antennapod/core/util/PodcastIndexTranscriptUtils.java

```diff
@@ -0,0 +1,65 @@
+package de.danoeh.antennapod.core.util;
+
+import android.os.NetworkOnMainThreadException;
+import android.util.Log;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import de.danoeh.antennapod.core.service.download.AntennapodHttpClient;
+import de.danoeh.antennapod.model.feed.FeedMedia;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class PodcastIndexTranscriptUtils {
+
+    private static final String TAG = "PodcastIndexTranscript";
+
+    public static String loadTranscriptFromUrl(String type, String url, boolean forceRefresh) {
+        StringBuffer str = new StringBuffer();
+        Thread downloadThread = new Thread() {
+            @Override
+            public void run() {
+                Response response = null;
+                try {
+                    Log.d(TAG, "Downloading transcript URL " + url.toString());
+                    Request request = new Request.Builder().url(url).build();
+                    response = AntennapodHttpClient.getHttpClient().newCall(request).execute();
+                    if (response.isSuccessful() && response.body() != null) {
+                        str.append(response.body().string());
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                } catch (NetworkOnMainThreadException e) {
+                    e.printStackTrace();
+                } finally {
+                    if (response != null) {
+                        response.close();
+                    }
+                }
+            }
+        };
+        downloadThread.start();
+        try {
+            downloadThread.join();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
```

Comment:
If this blocks the calling thread anyway, there is no need to open a new thread and join it. You can do the network directly on the calling thread.

Reply:
thanks for the catch